db/hints/manager.cc

```diff
@@ -1288,7 +1288,7 @@ void manager::rebalance_segments_for(
     }
 
     for (unsigned i = 0; i < smp::count && !segments_to_move.empty(); ++i) {
-        fs::path shard_path_dir(fs::path(hints_directory.c_str()) / seastar::format("{:d}", i).c_str() / ep.c_str());
+        fs::path shard_path_dir(hints_directory / seastar::format("{:d}", i).c_str() / ep.c_str());
```

Comment:
note to myself, we don't need to use `seastar::format()` here. `fmt::to_string()` would be better.